library/core/src/array/mod.rs

```diff
@@ -426,6 +427,47 @@ impl<T: Clone, const N: usize> Clone for [T; N] {
     }
 }
 
+#[unstable(feature = "random", issue = "130703")]
+impl<T: Random, const N: usize> Random for [T; N] {
+    default fn random(source: &mut (impl RandomSource + ?Sized)) -> Self {
+        from_fn(|_| T::random(source))
+    }
+}
+
+macro_rules! impl_random_for_integer_array {
+    ($t:ty) => {
+        #[unstable(feature = "random", issue = "130703")]
+        impl<const N: usize> Random for [$t; N] {
```

Comment:
Specializations should be hidden behind a private trait. https://std-dev-guide.rust-lang.org/policy/specialization.html

Alternatively you can avoid specialization by implementing a buffering random source wrapper as mentioned in https://github.com/rust-lang/rust/pull/136732#issuecomment-2645871365